pageserver/src/restore_local_repo.rs

```diff
@@ -180,6 +286,16 @@ pub fn import_timeline_wal(walpath: &Path, timeline: &dyn Timeline, startpoint:
     let mut segno = startpoint.segment_number(pg_constants::WAL_SEGMENT_SIZE);
     let mut offset = startpoint.segment_offset(pg_constants::WAL_SEGMENT_SIZE);
     let mut last_lsn = startpoint;
+
+    let checkpoint_bytes = timeline.get_page_at_lsn_nowait(ObjectTag::Checkpoint, startpoint)?;
+    let mut checkpoint = CheckPoint::decode(&checkpoint_bytes)?;
+    if checkpoint.nextXid.value == 0 {
```

Comment:
When is 'nextXid` invalid? Postgres always sets it to a valid value

Reply:
Zero value of nextXid is set by default constructor of CheckPoint struct. So this check is used to determine that CheckPoint needs to be initialized fro control file.

Reply:
Ok. I would suggest creating the CheckPoint struct from the control file immediately when the control file is imported. Then you could rely that it's always present. At a quick glance, `add_pgcontrol_file` doesn't currently seem to handle the case that the checkpoint object is missing. Maybe there is something else that ensures that it's been created by the time you get to `add_pgcontrol_file`, but it seems fragile.